electrode_models/nucleation_single_particle/initialize.py

```diff
@@ -0,0 +1,129 @@
+
+        # Store the species index of the Li ion in the Cantera object for the 
+        # electrolyte phase:
+        index_Li = elyte_obj.species_index(inputs['mobile-ion'])
+
+        # Electrode thickness and inverse thickness:
+        dy = inputs['thickness']
+        dyInv = 1/dy
+
+        # Phase volume fractions
+        eps_solid = inputs['eps_solid']
+        eps_elyte = 1 - eps_solid
+
+        # Electrode-electrolyte interface area, per unit geometric area.
+        # This calculation assumes spherical particles of a single radius, with 
+        # no overlap.
+        A_surf_ratio = (3*eps_solid*dy/inputs['r_p'])
+
+        # Inverse double layer capacitance, per unit interfacial area.
+        C_dl_Inv = 1/inputs['C_dl']
+
+        # Microstructure-based transport scaling factor, based on Bruggeman 
+        # coefficient of -0.5:
+        elyte_microstructure = eps_elyte**1.5
+        
+        # SV_offset specifies the index of the first SV variable for the 
+        # electrode (zero for anode, nVars_anode + nVars_sep for the cathode)
+        SV_offset = offset
+
+        # Determine the electrode capacity (Ah/m2)
+
+        # Max concentration of stored ion (intercalated Li)
+        # Save initial X
+        X_o = bulk_obj.X 
+        # Set object concentration to fully lithiated:
+        bulk_obj.X = inputs['stored-ion']['name']+':1.0' 
+        # Concentration of stored Li, per unit volume of intercalation phase:
+        C = bulk_obj[inputs['stored-ion']['name']].concentrations[0]
+        
+        capacity = (C*inputs['stored-ion']['charge']*ct.faraday
+                *inputs['eps_solid'])*inputs['thickness']/3600
+
+        # Return Cantera object composition to original value:
+        bulk_obj.X = X_o
+        
+        # Number of state variables: electrode potential, electrolyte 
+        # potential, electrode composition (n_species), electrolyte composition 
+        # (n_species)
+        nVars = 2 + bulk_obj.n_species + elyte_obj.n_species + inputs['buckets']
+
+        # Load the residual function and other required functions and store 
+        # them as methods of this class:
+        from .functions import residual, voltage_lim
+
+    # Set Cantera object state:
+    electrode.bulk_obj.electric_potential = inputs['phi_0']
+    # If the user provided an initial composition, use that, here:
+    if 'X_0' in inputs:
+        electrode.bulk_obj.TPX = (params['T'], params['P'], inputs['X_0'])
+    else:
+        electrode.bulk_obj.TP = params['T'], params['P']
+
+    electrode.elyte_obj.TP = params['T'], params['P']
+    electrode.surf_obj.TP = params['T'], params['P']
+    electrode.conductor_obj.TP = params['T'], params['P']
+
+    # Initialize the solution vector for the electrode domain:
+    SV = np.zeros([electrode.nVars])
+
+    # Set up pointers to specific variables in the solution vector:
+    electrode.SVptr = {}
+    electrode.SVptr['phi_ed'] = np.array([0])
```

Comment:
These pointers should reflect the variables listed above (2 electric potentials, elyte species concentrations, and histogram concentrations).